packages/lxd/src/file/pull.coffee.md

```diff
@@ -0,0 +1,139 @@
+# `nikita.lxc.file.pull`
+
+Pull files from containers.
+
+## Example
+
+```js
+const {$status} = await nikita.lxc.file.pull({
+  container: 'my_container',
+  source: './a_file',
+   target: `#{scratch}/a_file`
+})
+console.info(`File was pulled: ${$status}`)
+```
+
+## Todo
+
+* Ensure encrypting work as intended
+* Ensure gid and uid work as intended
+* Pull recursive directories
+* Detect name from lxd_source
+
+## Schema definitions
+
+    definitions =
+      config:
+        type: 'object'
+        properties:
+          'algo':
+            default: 'md5'
+            $ref: 'module://@nikitajs/core/lib/actions/fs/hash#/definitions/config/properties/algo'
+          'container':
+            $ref: 'module://@nikitajs/lxd/src/init#/definitions/config/properties/container'
+          'create_dirs':
+            type: 'boolean'
+            default: false
+            description: '''
+            Create any directories necessary.
+            '''
+          'gid':
+            type: ['integer', 'string']
+            description: '''
+            Set the file's gid on push, overwrite the `target` option.
+            '''
+          'lxd_source':
+            type: 'string'
+            description: '''
+            File source in the form of "[<remote>:]<container>/<path>",
+            overwrite the `source` option.
+            '''
+          'source':
+            type: 'string'
+            description: '''
+            File to pull in the form of "<path>".
+            '''
```

Comment:
Not sure what source is, not the lxd_source, not the target. The description doesn't really help.

Reply:
The source is basically the path of the file inside of the container. It can be, for instance `/root/file.txt` if the file that we want is at the root of our container.

The format of the command is: `lxc file pull remote/path_to_file path_to_destination` with the first argument being the container and the second being the local path on our machine.

Reply:
This has been modified.